update ISSUE_TEMPLATE folder files

cliffordfajardo opened the pull request:
## What is the change?
1. add `bug_report.md` to `bug-report.yml` to enable Github's form based issue template
   - https://youtu.be/qQE1BUkf2-s?t=23
2.  add `config.yml` file to help direct users to the helpful pages


## Motivation
- encourage's bug reporter's to put more care into their bug report before submission
- this may help maintainer's receive more detailed & higher quality bug report's
- adds helpful tips for user's during the process of creating a bug/issue report

## Demo of Change
- [`bug_report.yml` file](https://github.com/cliffordfajardo/nextui/blob/main/.github/ISSUE_TEMPLATE/bug_report.yml)
- [`feature-request.yml` file](https://github.com/cliffordfajardo/nextui/blob/main/.github/ISSUE_TEMPLATE/feature_request.yml)
- `config.yml` file


### Types of changes
- [X] Documentation Only

cliffordfajardo commented:
@jrgarciadev  
Feel free to make edit to my Github branch, github now has Codespaces enabled for nearly all repos on Github.com 🎉  
So you don't even need to run my code on your local computer since Codespaces allows you to edit in the browser now



My fork of nextui: https://github.com/cliffordfajardo/nextui

![CleanShot 2022-02-28 at 13 20 42@2x](https://user-images.githubusercontent.com/6743796/156060573-d9a1186a-1de8-4091-9939-9f06dbaf1bdc.png)

Dhananjay-JSR commented on .github/ISSUE_TEMPLATE/bug_report.yml:
```diff
@@ -0,0 +1,88 @@
+        This issue tracker is for reporting bugs found in `NextUI` (https://github.com/nextui-org/nextui/).
+        If you have a question about how to achieve something and are struggling, please post a question
+        inside of either of the following places:
+          - NextUI's Discussion's tab: https://github.com/nextui-org/nextui/discussions
+          - NextUI's Discord channel: https://foambubble.github.io/join-discord/g
+          
+
+        Before submitting a new bug/issue, please check the links below to see if there is a solution or question posted there already:
+         - NextUI's Issue's tab: https://github.com/nextui-org/nextui/pulls?q=is%3Apr+is%3Aopen+sort%3Aupdated-desc
+         - NextUI's closed issues tab: https://github.com/nextui-org/nextui/issues?q=is%3Aissue+sort%3Aupdated-desc+is%3Aclosed
+         - NextUI's Discussions tab: https://github.com/nextui-org/nextui/discussions
```
Same Goes here (wrap links as text) as it looks way more clean :- 

> also can you please replace discord with direct invite [link](https://discord.gg/9b6yyZKmH4) 

![image](https://user-images.githubusercontent.com/47073516/156115165-b1748a8d-0b42-4d09-840d-794acaf784b8.png)

**_Expected Output:-_**

![image](https://user-images.githubusercontent.com/47073516/156115205-98b9e7b6-b94b-48be-8967-7e61f9ed5184.png)

Dhananjay-JSR commented on .github/ISSUE_TEMPLATE/bug_report.yml:
```diff
@@ -0,0 +1,88 @@
+        For more information on the supported file image/file types and the file size limits, please refer
+        to the following link: https://docs.github.com/en/github/writing-on-github/working-with-advanced-formatting/attaching-files
+      placeholder: |
+        You can drag your video or image files inside of this editor ↓
```
it would be better to have a dropdown to select browser where bug was discovered

![image](https://user-images.githubusercontent.com/47073516/156114252-b20d8950-11d0-4385-8623-875c8a3ff118.png)

**_OUTPUT:-_**

![image](https://user-images.githubusercontent.com/47073516/156114341-7a9a4476-9a20-4358-a735-69a516d6ca27.png)

cliffordfajardo replied:
Nice - i like this

Dhananjay-JSR commented on .github/ISSUE_TEMPLATE/bug_report.yml:
```diff
@@ -0,0 +1,88 @@
+    attributes:
+      label: Additional context
+      description: | 
+        Add any other context about the problem here.
```
I think it would be great to add [CODE OF CONDUCT](https://github.com/nextui-org/nextui/blob/main/CODE_OF_CONDUCT.md) check box 

![image](https://user-images.githubusercontent.com/47073516/156114978-1908f065-3491-4075-812b-71fdf7e08a92.png)

Output:- 

![image](https://user-images.githubusercontent.com/47073516/156115030-1050e913-0a27-4dc7-af85-2f07f892bfd4.png)

Dhananjay-JSR commented on .github/ISSUE_TEMPLATE/feature_request.yml:
```diff
@@ -0,0 +1,47 @@
+name: Feature request
+description:  |
```
Add title and labels tag as mentioned above
`title: "[Feature Request] <Title>"`
`labels: [feature request]`

![image](https://user-images.githubusercontent.com/47073516/156115519-fd2c72d2-7153-4d74-adb7-be30140b0a8b.png)

Dhananjay-JSR commented on .github/ISSUE_TEMPLATE/feature_request.yml:
```diff
@@ -0,0 +1,47 @@
+        For more information on the supported file image/file types and the file size limits, please refer
+        to the following link: https://docs.github.com/en/github/writing-on-github/working-with-advanced-formatting/attaching-files
+      placeholder: |
+        You can drag your video or image files inside of this editor ↓
```
Add Code of Conduct Checkbox here too

![image](https://user-images.githubusercontent.com/47073516/156115817-e64c7aa7-6b07-4e82-85d7-31beb0dffdb1.png)

OUTPUT:- 


![image](https://user-images.githubusercontent.com/47073516/156115876-bb238804-fb43-4eab-b3f4-92d0bf425d7c.png)

Dhananjay-JSR commented on .github/ISSUE_TEMPLATE/feature_request.yml:
```diff
@@ -0,0 +1,47 @@
+      description: |
+        If applicable, add screenshots or a video to help explain your problem.
+        For more information on the supported file image/file types and the file size limits, please refer
+        to the following link: https://docs.github.com/en/github/writing-on-github/working-with-advanced-formatting/attaching-files
```
**_Same as in Bug Report_**

![image](https://user-images.githubusercontent.com/47073516/156115698-48d151db-b2f5-4a1d-8b73-81267fc2298f.png)

cliffordfajardo replied:
I'm a bit reluctant to make this a hyper link (not opposed) I think one benefit of having  the long URL is that i'ts harder for people to miss - and the goal is to reduce friction for maintainers and make resources easily accessible (hard to miss) for contributors

Dhananjay-JSR commented on .github/ISSUE_TEMPLATE/bug_report.yml:
```diff
@@ -0,0 +1,88 @@
+      description: |
+        If applicable, add screenshots or a video to help explain your problem.
+        For more information on the supported file image/file types and the file size limits, please refer
+        to the following link: https://docs.github.com/en/github/writing-on-github/working-with-advanced-formatting/attaching-files
```
this can be further simplify into 

![image](https://user-images.githubusercontent.com/47073516/156115641-655c16f8-340e-483b-9fe3-cc3218482822.png)

cliffordfajardo replied:
I'm a bit reluctant to make this a hyper link (not opposed) I think one benefit of having the long URL is that i'ts harder for people to miss - and the goal is to reduce friction for maintainers and make resources easily accessible (hard to miss) for contributors

Dhananjay-JSR reviewed (changes requested):
Hi @cliffordfajardo ! Thank you so much for being willing to contribute 😊Would you be open to updating your PR with the changes described ?

cliffordfajardo commented on .github/ISSUE_TEMPLATE/bug_report.yml:
```diff
@@ -0,0 +1,88 @@
+name: 'Bug report'
+description: Create a report to help us improve
```
Nice - i didn't know about labels and title defaults - thanks! 🙏

cliffordfajardo commented:
Hi @Dhananjay-JSR - thank your for the valuable PR feedback. I learned quite a bit of the new features thanks to your suggesstions.

Changes made:
- added Code of Conduct
- fixed typos
- added field for selecting browser from dropdown list


### Follow Up Question from your original pull request feedback
I'm a bit reluctant to make the hyper links shorter for the some of the links inside the description labels since Github renders those very small & its easy to miss those links

![CleanShot 2022-03-01 at 09 11 36@2x](https://user-images.githubusercontent.com/6743796/156216010-4867d4d8-37b5-4f1c-9078-064ab752c519.png)



I believe one benefit of having the long URL (in places that is easier to miss) is that its much harder for people to miss the useful links in this context. Typically a user might be feeling rushed or frustrated when submitting a bug report or pull request and they might glance over the details.

The long explicit urls helps reduce friction for maintainers and make resources/tips for contributors a bit more easily accessible (hard to miss)

@jrgarciadev - any thoughts?


Not that this matters too much (view it more as a data point), other repos like solidJS and react-query also have the longer explicit URLs
- [react-query bug_report.yml](https://github.com/tannerlinsley/react-query/blob/master/.github/ISSUE_TEMPLATE/bug_report.yml) file

- [solidjs bug_report.yml](https://github.com/solidjs/solid/blob/main/.github/ISSUE_TEMPLATE/bug_report.yml) file

jrgarciadev commented:
Hey, @cliffordfajardo this looks amazing, thank you so much, just one thing, could you please remove the line that says `Code of conduct`?

cliffordfajardo commented:
@jrgarciadev -
1.are you saying remove the code of conduct from the `feature_request.yml` and `bug_report.yml`?
2.can you point to the specific line or file?

jrgarciadev commented on .github/ISSUE_TEMPLATE/bug_report.yml:
```diff
@@ -0,0 +1,112 @@
+      label: Additional context
+      description: | 
+        Add any other context about the problem here.
+  - type: checkboxes
```
@cliffordfajardo can you please remove this section?

cliffordfajardo replied:
Updated

Dhananjay-JSR commented on .github/ISSUE_TEMPLATE/bug_report.yml:
```diff
@@ -0,0 +1,98 @@
+        If you have a question about how to achieve something and are struggling, please post a question
+        inside of either of the following places:
+          - NextUI's [Discussion's tab](https://github.com/nextui-org/nextui/discussions)
+          - NextUI's [Discord channel](https://foambubble.github.io/join-discord/g)
```
Hey @cliffordfajardo can you please update Discord Link , I did mentioned this in last review report looks like you forget to fix it 😄

cliffordfajardo replied:
WHoops - apologies - the code has been merged but a new PR can be created

Dhananjay-JSR replied:
Do you want to open a new PR ?

jrgarciadev replied:
Hey @cliffordfajardo @Dhananjay-JSR sorry I didn't notice that, fixed it now 👍🏻
